src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -0,0 +1,236 @@
+  IL_0022:  ret
+}");
+            verifier.VerifyIL("C.Test2", @"
+{
+  // Code size       64 (0x40)
+  .maxstack  2
+  .locals init (A? V_0,
+                B? V_1,
+                int? V_2)
+  IL_0000:  ldarg.0
+  IL_0001:  ldfld      ""A? S.Prop1""
+  IL_0006:  stloc.0
+  IL_0007:  ldloca.s   V_0
+  IL_0009:  call       ""bool A?.HasValue.get""
+  IL_000e:  brfalse.s  IL_003e
+  IL_0010:  ldloca.s   V_0
+  IL_0012:  call       ""A A?.GetValueOrDefault()""
+  IL_0017:  ldfld      ""B? A.Prop2""
+  IL_001c:  stloc.1
+  IL_001d:  ldloca.s   V_1
+  IL_001f:  call       ""bool B?.HasValue.get""
+  IL_0024:  brfalse.s  IL_003e
+  IL_0026:  ldloca.s   V_1
+  IL_0028:  call       ""B B?.GetValueOrDefault()""
+  IL_002d:  ldfld      ""int? B.Prop3""
+  IL_0032:  stloc.2
+  IL_0033:  ldloca.s   V_2
+  IL_0035:  call       ""bool int?.HasValue.get""
+  IL_003a:  ldc.i4.0
+  IL_003b:  ceq
+  IL_003d:  ret
+  IL_003e:  ldc.i4.0
+  IL_003f:  ret
+}");
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_02()
+        {
+            var program = @"
+class C
+{
+    public C Prop1 { get; set; }
+    public C Prop2 { get; set; }
+
+    public static void Main(string[] args)
+    {        
+        _ = new C() is { Prop1: null } and { Prop1.Prop2: null };
+        _ = new C() is { Prop1: null, Prop1.Prop2: null };
+    }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendPropertyPattern, options: TestOptions.ReleaseExe);
+            compilation.VerifyDiagnostics(
+                // (9,13): error CS8518: An expression of type 'C' can never match the provided pattern.
+                //         _ = new C() is { Prop1: null } and { Prop1.Prop2: null };
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "new C() is { Prop1: null } and { Prop1.Prop2: null }").WithArguments("C").WithLocation(9, 13),
+                // (10,13): error CS8518: An expression of type 'C' can never match the provided pattern.
+                //         _ = new C() is { Prop1: null, Prop1.Prop2: null };
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "new C() is { Prop1: null, Prop1.Prop2: null }").WithArguments("C").WithLocation(10, 13)
+                );
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_03()
+        {
+            var program = @"
+using System;
+class C
+{
+    public C Prop1 { get; set; }
+    public C Prop2 { get; set; }
+
+    public static void Main(string[] args)
+    {        
+        var result = new C() switch
+        {
+            {Prop1: null} => 1,
+            {Prop1.Prop2: null} => 2,
+            _ => -1,
+        };
+        Console.Write(result);
+    }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendPropertyPattern, options: TestOptions.ReleaseExe);
+            compilation.VerifyDiagnostics();
+            var verifier = CompileAndVerify(compilation);
```

Comment:
Please verify some output in CompileAndVerify tests.
In this case, it would be good to have `Prop1` have a side-effect (print to console) as extra illustration that it is evaluated only once.